<Issue id="15446" key="FAB-2873" number="2873" project="10002" reporter="harsha544" creator="harsha544" type="10004" summary="Failing behave scenarios/features on ppc64le" priority="3" resolution="10000" status="6" created="2017-03-24 11:59:31.0" updated="2018-07-20 14:12:03.0" resolutiondate="2017-04-14 00:36:40.0" votes="0" watches="6" workflowId="37837"> <description><! CDATA Evaluating behave tests on master code, the 4 x CouchDB scenario of bootstrap.feature is failing https://github.com/hyperledger/fabric/blob/master/bddtests/features/bootstrap.feature#L283  Given user "dev0Org0" gives "genesisBlockForMyNewChannel" to user "dev0Org1"                                                                                                                                                                     # steps/bootstrap_impl.py:331 0.000s When user "dev0Org0" using cert alias "dev0Org0App1" requests to join channel using genesis block "genesisBlockForMyNewChannel" on peers with result "joinChannelResult"                                                                         # steps/bootstrap_impl.py:256 0.325s | Peer  | | peer0 | | peer1 | Traceback (most recent call last): File "/root/Envs/behave_venv2/local/lib/python2.7/site-packages/behave/model.py", line 1456, in run match.run(runner.context) File "/root/Envs/behave_venv2/local/lib/python2.7/site-packages/behave/model.py", line 1903, in run self.func(context, *args, **kwargs) File "steps/bootstrap_impl.py", line 274, in step_impl resultsDict =  dict(zip(endorsers,  respFuture.result() for respFuture in proposalResponseFutures )) File "/root/Envs/behave_venv2/local/lib/python2.7/site-packages/grpc/_channel.py", line 295, in result raise self _Rendezvous: <_Rendezvous of RPC that terminated with (StatusCode.UNAVAILABLE, Connect Failed)>  Captured stdout: Will copy gensisiBlock over at this point ipAddress in getABStubForComposeService == 172.20.0.3 Returning GRPC for address: 172.20.0.3 dev0Org0 received reply: status: SUCCESS , counter = 1 Done Start and end = 0/0  SeekInfo = start { specified { } } stop { specified { } } behavior: FAIL_IF_NOT_READY   Done receiving blocks Returning GRPC for address: 172.20.0.7 Returning GRPC for address: 172.20.0.9  ```  docker ps -a shows, none of the containers are in exit state And here's what I see inside the newly spawned peer-container :-  ``` 2017-03-23 14:54:26.098 UTC  couchdb  handleRequest -> DEBU 010 HTTP Request: GET / HTTP/1.1 | Host: couchdb2:5984 | User-Agent: Go-http-client/1.1 | Accept: multipart/related | Accept-Encoding: gzip |  | 2017-03-23 14:54:26.101 UTC  couchdb  handleRequest -> WARN 011 Retrying couchdb request in 125ms. Attempt:1  Error:Get http://couchdb2:5984/: dial tcp 172.20.0.2:5984: getsockopt: connection refused 2017-03-23 14:54:26.229 UTC  couchdb  handleRequest -> WARN 012 Retrying couchdb request in 250ms. Attempt:2  Error:Get http://couchdb2:5984/: dial tcp 172.20.0.2:5984: getsockopt: connection refused 2017-03-23 14:54:26.482 UTC  couchdb  handleRequest -> WARN 013 Retrying couchdb request in 500ms. Attempt:3  Error:Get http://couchdb2:5984/: dial tcp 172.20.0.2:5984: getsockopt: connection refused 2017-03-23 14:54:26.985 UTC  couchdb  handleRequest -> WARN 014 Retrying couchdb request in 1s. Attempt:4  Error:Get http://couchdb2:5984/: dial tcp 172.20.0.2:5984: getsockopt: connection refused 2017-03-23 14:54:27.989 UTC  couchdb  handleRequest -> WARN 015 Retrying couchdb request in 2s. Attempt:5  Error:Get http://couchdb2:5984/: dial tcp 172.20.0.2:5984: getsockopt: connection refused 2017-03-23 14:54:30.078 UTC  couchdb  handleRequest -> WARN 016 Retrying couchdb request in 4s. Attempt:6  Error:Get http://couchdb2:5984/: dial tcp 172.20.0.2:5984: getsockopt: connection refused 2017-03-23 14:54:34.082 UTC  couchdb  handleRequest -> WARN 017 Retrying couchdb request in 8s. Attempt:7  Error:Get http://couchdb2:5984/: dial tcp 172.20.0.2:5984: getsockopt: connection refused 2017-03-23 14:54:42.086 UTC  couchdb  handleRequest -> WARN 018 Retrying couchdb request in 16s. Attempt:8  Error:Get http://couchdb2:5984/: dial tcp 172.20.0.2:5984: getsockopt: connection refused 2017-03-23 14:54:58.090 UTC  couchdb  handleRequest -> WARN 019 Retrying couchdb request in 32s. Attempt:9  Error:Get http://couchdb2:5984/: dial tcp 172.20.0.2:5984: getsockopt: connection refused 2017-03-23 14:55:30.093 UTC  couchdb  handleRequest -> WARN 01a Retrying couchdb request in 1m4s. Attempt:10  Error:Get http://couchdb2:5984/: dial tcp 172.20.0.2:5984: getsockopt: connection refused panic: Error in instantiating ledger provider: Unable to connect to CouchDB, check the hostname and port: Get http://couchdb2:5984/: dial tcp 172.20.0.2:5984: getsockopt: connection refused  goroutine 1  running : panic(0x108c4da0, 0xc420232060) /opt/go/src/runtime/panic.go:500 +0x3a8 github.com/hyperledger/fabric/core/ledger/ledgermgmt.initialize() /opt/gopath/src/github.com/hyperledger/fabric/core/ledger/ledgermgmt/ledger_mgmt.go:59 +0x278 github.com/hyperledger/fabric/core/ledger/ledgermgmt.Initialize.func1() /opt/gopath/src/github.com/hyperledger/fabric/core/ledger/ledgermgmt/ledger_mgmt.go:47 +0x24 sync.(*Once).Do(0x10de11c0, 0x10a4d5a0) /opt/go/src/sync/once.go:44 +0x190 github.com/hyperledger/fabric/core/ledger/ledgermgmt.Initialize() /opt/gopath/src/github.com/hyperledger/fabric/core/ledger/ledgermgmt/ledger_mgmt.go:48 +0x3c github.com/hyperledger/fabric/peer/node.serve(0xc42024f890, 0x0, 0x1, 0x0, 0x0) /opt/gopath/src/github.com/hyperledger/fabric/peer/node/start.go:94 +0x48 github.com/hyperledger/fabric/peer/node.glob..func1(0x10d8a3a0, 0xc42024f890, 0x0, 0x1, 0x0, 0x0) /opt/gopath/src/github.com/hyperledger/fabric/peer/node/start.go:82 +0x44 github.com/hyperledger/fabric/vendor/github.com/spf13/cobra.(*Command).execute(0x10d8a3a0, 0xc42024f870, 0x1, 0x1, 0x0, 0x0) /opt/gopath/src/github.com/hyperledger/fabric/vendor/github.com/spf13/cobra/command.go:599 +0x4c8 github.com/hyperledger/fabric/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x10d8ac20, 0x10d8a3a0, 0x0, 0x0) /opt/gopath/src/github.com/hyperledger/fabric/vendor/github.com/spf13/cobra/command.go:689 +0x3f4 github.com/hyperledger/fabric/vendor/github.com/spf13/cobra.(*Command).Execute(0x10d8ac20, 0x0, 0x0) /opt/gopath/src/github.com/hyperledger/fabric/vendor/github.com/spf13/cobra/command.go:648 +0x34 main.main() /opt/gopath/src/github.com/hyperledger/fabric/peer/main.go:111 +0x6c0 (behave_venv2) root@sys-86315:/opt/gopath/src/github.com/hyperledger/fabric/bddtests#  ></description> </Issue>
